README


This archive contains the distribution App-PerlWatcher-Engine,
version 0.20:

  Desktop nagios-inspired events watcher/notifier

This software is copyright (c) 2014 by Ivan Baidakou.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.




Changes
0.20      2014-01-07 22:04:03 Europe/Minsk

0.20      2013-12-29 18:37:44 Europe/Minsk
    [API] WatcherMemory has been generalized into Memory
    [API] EventItems inside emitted by Watcher Status, now can be the previous
          ones, if they match by *content* in previous Status emission
    [API] EventItem is now Memorizable

0.19      2013-12-18 02:06:43 Europe/Minsk
    Now, it should pass all tests on windows. (No changes since 0.18_3
    except added Gtk2+ Frontend Screenshot)

0.18_3    2013-12-14 16:54:27 Europe/Minsk
    [API] Migrate from Path::Class to Path::Tiny usage
    [Fix] Let t/101-engine-with-ping.t pass tests on Windows by using the recent
    Path::Tiny version

0.18_2    2013-12-01 18:59:38 Europe/Minsk
    [API] Add 'poll_callback' for watcher to notify UI that some resource is been 
    polled
    [Fix] Let GenericExecutor work on Win32

0.18      2013-11-16 19:43:18 Europe/Minsk
    [API] Extract Rss watcher into separate package 
          (Now, different RSS and ATOM formats are supported)
    [Fix] Make tests more tolerant on non-linux systems

0.17_1    2013-11-14 00:02:30 Europe/Minsk
    Minor changes passing tests on Windows
    (Thanks to Alexandr Ciornii!)

0.17      2013-11-13 03:24:44 Europe/Minsk
    No changes since 0.16_5

0.16_5    2013-11-02 18:34:51 Europe/Minsk
    Migrate from Devel::Comments to Smart::Comments

0.16_4    2013-11-02 13:45:19 Europe/Minsk
    [Fix] Remove Test::Pod from tests to fix building

0.16_3    2013-10-27 12:52:59 Europe/Minsk
    [Fix] Minor fixes in distribution, e.g. add git repository url

0.16_2    2013-09-13 00:24:15 Europe/Minsk
    [Fix] Once again fix watcher's unique_id generation algorithm.
    [API] GenericExecutor Watcher now has 'beautifyer' closure, which is been
    applied for every line of filtered output

0.16_1    2013-09-10 00:16:59 Europe/Minsk
    [Fix] Do not include POSIX in dependencies.

0.16      2013-09-10 00:04:16 Europe/Minsk
    Added GenericExecutor Watcher to periodically execute arbitrary command
    	  and watch/view it's output
    The watcher's unique_id generation algorithm changed to handle correctly
    	  deep configuration structures.
    [API] Watcher Memory now stores 'active' state of Watcher
    [API] Watcher role now requires 'callback' parameter in instance construction,
    	  'start' method does not takes the callback as 2nd parameter any more

0.15      2013-08-31 23:01:57 Europe/Minsk
    Ping Watcher now supports ICMP ping (via Net::Ping::External) as well as TCP knock 
    [API] Watcher has 'build_watcher_guard' method, which is responsible for constructing 
          AnyEvent-based guard (AnyEvent->timer etc.) 
    [API] Watcher role now has 'Describable' role
    [API] Watcher role now has 'force_poll' and 'start' methods
    [API] Describable role now has 'describe', 'describer' and 'description' methods

0.14_01   2013-08-26 20:23:25 Europe/Minsk
    [fix] Build on older ExtUtils::MakeMaker (RT #88080)
    FileTail watcher was separated as Linux-specific into App::PerlWatcher::Watcher::FileTail

0.14      2013-08-24 17:27:29 Europe/Minsk
    Relased to CPAN
    Make the default/sample config more adequate for default installation
    [fix] File watcher: don't die, if tracked file can't be found
    [fix] Added missing dependencies

0.13      2013-08-18 17:13:20 Europe/Minsk
    Added POD documentation
    Added Openable role to make RSS-news items openable in browser
    Finalize Mooification 
    [API change] Engine does not takes an backend_id any more, provide the backend instance
    	 	 for that. Added Backend role for that.
    [API change] The available leves are exported via App::PerlWatcher::Levels now

0.12      2013-07-30 17:26:47 Europe/Minsk
    Updated Sortable to 2.45
    Mirgration to Moo instead of bare perl v5 OOP
    [config change] FileTail: lines -> lines_number
    [config change] Rss: _items_count-> items_number

0.11      2013-07-25 12:16:37 Europe/Minsk
    The build was adopted to travis-ci
    Persists application state (statuses with levels) between invocations
    Added Weather watcher
    [fix] Build correctly on perl v5.12

0.10      2013-07-08 01:05:47 Europe/Minsk
    Use Dist::Zilla's @git plugins bundle

0.09      2013-07-08 00:49:44 Europe/Minsk
    Separation Gtk frontend from main engine

0.08      2013-06-27 23:44:24 Europe/Minsk
  [Improvements]
      - Moved to using Dist::Zilla instead of old-school ExtUtils::MakeMaker



Makefile.PL

use strict;
use warnings;

use 5.012;

use ExtUtils::MakeMaker 6.72;

use File::ShareDir::Install;
install_share dist => "share";


my %WriteMakefileArgs = (
  "ABSTRACT" => "Desktop nagios-inspired events watcher/notifier",
  "AUTHOR" => "Ivan Baidakou <dmol\@gmx.com>",
  "BUILD_REQUIRES" => {},
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.72",
    "File::ShareDir::Install" => "0.03"
  },
  "DISTNAME" => "App-PerlWatcher-Engine",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "App::PerlWatcher::Engine",
  "PREREQ_PM" => {
    "AnyEvent" => "7.04",
    "AnyEvent::HTTP" => "2.15",
    "Class::Load" => "0.20",
    "Data::Dump" => "1.22",
    "File::ReadBackwards" => "1.05",
    "File::ShareDir::ProjectDistDir" => "v0.4.4",
    "Function::Parameters" => "1.0401",
    "IPC::Cmd" => "0.90",
    "IPC::Run3" => "0.045",
    "List::MoreUtils" => "0.33",
    "Moo" => "1.003000",
    "Net::Ping::External" => "0.13",
    "Path::Tiny" => "0.049",
    "Smart::Comments" => "1.000004",
    "Storable" => "2.45",
    "Type::Tiny" => "0.038",
    "URI" => "1.60",
    "XML::XPath" => "1.13",
    "aliased" => "0.31"
  },
  "TEST_REQUIRES" => {
    "AnyEvent::HTTPD" => "0.93",
    "File::Spec" => 0,
    "File::Temp" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => 0,
    "Test::TCP" => "1.26",
    "Test::Warnings" => "0.013"
  },
  "VERSION" => "0.20",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "AnyEvent" => "7.04",
  "AnyEvent::HTTP" => "2.15",
  "AnyEvent::HTTPD" => "0.93",
  "Class::Load" => "0.20",
  "Data::Dump" => "1.22",
  "File::ReadBackwards" => "1.05",
  "File::ShareDir::ProjectDistDir" => "v0.4.4",
  "File::Spec" => 0,
  "File::Temp" => 0,
  "Function::Parameters" => "1.0401",
  "IO::Handle" => 0,
  "IPC::Cmd" => "0.90",
  "IPC::Open3" => 0,
  "IPC::Run3" => "0.045",
  "List::MoreUtils" => "0.33",
  "Moo" => "1.003000",
  "Net::Ping::External" => "0.13",
  "Path::Tiny" => "0.049",
  "Smart::Comments" => "1.000004",
  "Storable" => "2.45",
  "Test::More" => 0,
  "Test::TCP" => "1.26",
  "Test::Warnings" => "0.013",
  "Type::Tiny" => "0.038",
  "URI" => "1.60",
  "XML::XPath" => "1.13",
  "aliased" => "0.31"
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);

{
package
MY;
use File::ShareDir::Install qw(postamble);
}


